code/iaas/healthcheck/src/main/java/io/cattle/iaas/healthcheck/service/impl/HealthcheckServiceImpl.java

```diff
@@ -297,21 +297,23 @@ protected void createHealthCheckHostMaps(HealthcheckInstanceType instanceType, l
         availableActiveHostIds.removeAll(allocatedActiveHostIds);
         requiredNumber = requiredNumber - allocatedActiveHostIds.size();
         Collections.shuffle(availableActiveHostIds);
-
+   
         // place inferiorHostId to the end of the list
         if (inferiorHostId != null) {
-            if (availableActiveHostIds.contains(inferiorHostId)) {
-                availableActiveHostIds.remove(inferiorHostId);
-                if (availableActiveHostIds.isEmpty() && allocatedActiveHostIds.isEmpty()) {
+                if (availableActiveHostIds.contains(inferiorHostId)) {
+                    availableActiveHostIds.remove(inferiorHostId);
+                }
+                if(!allocatedActiveHostIds.contains(inferiorHostId)) {
```

Comment:
I don't think this condition can be true at this point; so we can just remove, and put it back

Reply:
we don't want id to be considered available if it's already allocated?

Reply:
sorry, read `allocatedActiveHostIds` as `availableActiveHostIds`